src/main/java/seedu/address/logic/commands/AddSkillCommand.java

```diff
@@ -0,0 +1,193 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SKILL;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_COURSE_MATES;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.coursemate.CourseMate;
+import seedu.address.model.coursemate.QueryableCourseMate;
+import seedu.address.model.coursemate.exceptions.CourseMateNotFoundException;
+import seedu.address.model.skill.Skill;
+
+/**
+ * Adds a courseMate to the contact list.
+ */
+public class AddSkillCommand extends Command {
+
+    public static final String COMMAND_WORD = "add-skill";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds skills to a coursemate. "
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_SKILL + " SKILL "
+            + "[" + PREFIX_SKILL + " SKILL]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_SKILL + " Python "
+            + PREFIX_SKILL + " Java";
+
+    public static final String MESSAGE_SUCCESS = "New skills added";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_COURSE_MATE = "This courseMate already exists in the contact list";
```

Comment:
To be solved later: I think all of us are using different capitalizations of coursemate. Need to be consistent.